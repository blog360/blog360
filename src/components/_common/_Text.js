import React from 'react';
import { Text } from 'react-vr';

function _Text(props) {
  const { text, styles } = props;
  return (
    <Text style={ styles }>{text}</Text>
  );
};

export default _Text;