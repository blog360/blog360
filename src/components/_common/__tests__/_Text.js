import React from 'react';
import { Text, StyleSheet } from 'react-vr';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import _Text from '../_Text';

test('_Text renders a _Text component', () => {
  const tree = renderer.create(<_Text />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('_Text renders a Text component', () => {
  const wrapper = shallow(<_Text text="Hello world!" styles={stylesheet.subtitle} />);
  console.log(wrapper.instance().props.text);
  expect(wrapper.instance().props.text).toBe('Hello world!');
});


const stylesheet = StyleSheet.create({
  subtitle: {
    fontSize: 0.2
  },
});

