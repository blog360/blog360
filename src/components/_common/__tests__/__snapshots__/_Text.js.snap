// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_Text renders a _Text component 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  isOnLayer={false}
  style={undefined}
/>
`;

exports[`_Text renders a Text component 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  isOnLayer={false}
  style={undefined}
/>
`;
